.github/workflows/release.yml

```diff
@@ -2,28 +2,37 @@ name: Deploy Main
 on:
   release:
     types: [published]
+
+env:
+  PYTHON_VERSION: "3.12"
+
 jobs:
   deploy-pypi:
     runs-on: ubuntu-latest
-    environment: deployment
     permissions:
       id-token: write # IMPORTANT: this permission is mandatory for trusted publishing
 
     steps:
       - name: Checkout Repository
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+          fetch-tags: true
 
       - name: Install Python
-        uses: actions/setup-python@v3
+        uses: actions/setup-python@v5
         with:
-          python-version: "3.10"
+          cache: 'pip'
+          cache-dependency-path: |
+            **/setup.py
```

Comment:
do we need this now that we've deprecated setup.py?

Reply:
probably not, and uv is faster than caching (!)